internal/cloud/gcp/oss/helpers.go

```diff
@@ -0,0 +1,262 @@
+package oss
```

Comment:
This is a bag of helpers from the terraform google provider.

It uses a Mozilla license: https://github.com/hashicorp/terraform-provider-google/blob/main/LICENSE

https://fossa.com/blog/open-source-software-licenses-101-mozilla-public-license-2-0/

+ Can I put this code here?
+ I'm mostly interested in the `mergePolicies` code
 
I previously implemented merge on my own but hadn't taken iam conditions into account yet.

Reply:
My read of this clause "Sublicense binaries. A developer may place binaries under a different license in the context of an aggregate work." Says we can use the code if the original package is not dumped into our code.

Interested in Cory and Jake's opinion

Reply:
Yeah that's fair.

I asked @jaketf about this a few days ago and his advice was to put it in a separate folder like this but _with_ the Apache license but admittedly hasn't worked with this license much before.

@coryodaniel thoughts?